src/main/java/org/prebid/server/bidder/improvedigital/ImprovedigitalBidder.java

```diff
@@ -67,6 +74,47 @@ public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request
         return Result.withValues(httpRequests);
     }
 
+    private ExtUser getAdditionalConsentProvidersUserExt(BidRequest request) {
+        if (request.getUser().getExt() == null) {
+            return null;
+        }
+        final ExtUser extUser = request.getUser().getExt();
+        final String consentedProviders = validateConsentedProvidersSettings(extUser.getConsentedProvidersSettings());
+
+        if (StringUtils.isBlank(consentedProviders)) {
+            return null;
+        }
+
+        final String[] arrayOfSplitString = StringUtils.substringAfterLast(consentedProviders, "~")
+                .split(REGEX_SPLIT_STRING_BY_DOT);
+
+        return fillExtUser(extUser, arrayOfSplitString);
```

Comment:
Make copy of `extUser` to avoid mutation.
For example: `extUset.toBuilder().build()`

Reply:
Sorry, missed up that `FlexibleExtension` don't use `@SuperBuilder`. For now we can't to do this with `extUset.toBuilder().build()`